database/models.py

```diff
@@ -18,9 +18,9 @@ class Sample(Base):
 
     id = Column(Integer, primary_key=True, nullable=False)
 
-    patient_id = Column(Integer, ForeignKey("patient.id"), nullable=True)
-    batch_id = Column(Integer, ForeignKey("batch.id"), nullable=False)
-    cohort_id = Column(String, ForeignKey("cohort.id"), nullable=False)
+    patient_id = Column(Integer, ForeignKey("patient.id", ondelete="CASCADE"), nullable=True)
+    batch_id = Column(Integer, ForeignKey("batch.id", ondelete="CASCADE"), nullable=False)
+    cohort_id = Column(String, ForeignKey("cohort.id", ondelete="CASCADE"), nullable=False)
```

Comment:
what does the `ondelete="CASCADE"` do?

Reply:
It means if the parent table (the table with the corresponding primary key to those foreignkeys) was deleted, then these children tables will also be deleted (cascade) - and in sqlalchemy, the parent table is considered where the relationship was defined. It's also possible to delete children and cascade that up to parents, which are the changes to the Cohort/Batch tables

Reply:
oh yep okay i get you!